File: apps/macos/Sources/CueWeaveMac/LocalSettings.swift
import Foundation

enum ModelDefaults {
    static let openRouter = "google/gemini-3.8-flash"
    static let aiStudio = "gemini-3.8-flash"

    fileprivate static let legacyOpenRouter = "google/gemini-3.7-flash"
    fileprivate static let legacyAIStudio = "gemini-3.7-flash"
}

struct LocalSettings: Codable, Equatable {
    var alignmentProvider: String?
    var openRouterAPIKey: String?
    var aiStudioAPIKey: String?
    var openRouterModel: String?
    var aiStudioModel: String?
    var uiLanguage: String?

    mutating func migrateLegacyModelDefaults() {
        if openRouterModel == ModelDefaults.legacyOpenRouter {
            openRouterModel = ModelDefaults.openRouter
        }
        if aiStudioModel == ModelDefaults.legacyAIStudio {
            aiStudioModel = ModelDefaults.aiStudio
        }
    }
}

enum LocalSettingsStore {
    static let configURL: URL = FileManager.default
        .urls(for: .applicationSupportDirectory, in: .userDomainMask)[0]
        .appendingPathComponent("CueWeave", isDirectory: true)
        .appendingPathComponent("config.json")

    static func load(from url: URL = configURL) -> LocalSettings {
        guard let data = try? Data(contentsOf: url),
              var settings = try? JSONDecoder().decode(LocalSettings.self, from: data)
        else { return LocalSettings() }
        settings.migrateLegacyModelDefaults()
        return settings
    }

    static func save(_ settings: LocalSettings, to url: URL = configURL) throws {
        let directory = url.deletingLastPathComponent()
        try FileManager.default.createDirectory(
            at: directory,
            withIntermediateDirectories: true,
            attributes: [.posixPermissions: 0o700]
        )
        try FileManager.default.setAttributes(
            [.posixPermissions: 0o700],
            ofItemAtPath: directory.path
        )
        let encoder = JSONEncoder()
        encoder.outputFormatting = [.prettyPrinted, .sortedKeys]
        try encoder.encode(settings).write(to: url, options: .atomic)
        try FileManager.default.setAttributes(
            [.posixPermissions: 0o600],
            ofItemAtPath: url.path
        )
    }
}

File: apps/windows/CueWeave.Windows/Models/LocalSettings.cs
namespace CueWeave.WinUI.Models;

public static class ModelDefaults
{
    public const string OpenRouter = "google/gemini-3.8-flash";
    public const string AiStudio = "gemini-3.8-flash";
    internal const string LegacyOpenRouter = "google/gemini-3.7-flash";
    internal const string LegacyAiStudio = "gemini-3.7-flash";
}

public sealed class LocalSettings
{
    public string AlignmentProvider { get; set; } = "openrouter";
    public string OpenRouterApiKey { get; set; } = "";
    public string AiStudioApiKey { get; set; } = "";
    public string OpenRouterModel { get; set; } = ModelDefaults.OpenRouter;
    public string AiStudioModel { get; set; } = ModelDefaults.AiStudio;
    public string UiLanguage { get; set; } = "system";

    public void MigrateLegacyModelDefaults()
    {
        if (OpenRouterModel == ModelDefaults.LegacyOpenRouter)
            OpenRouterModel = ModelDefaults.OpenRouter;
        if (AiStudioModel == ModelDefaults.LegacyAiStudio)
            AiStudioModel = ModelDefaults.AiStudio;
    }
}
